src/api.hpp

```diff
@@ -154,6 +191,33 @@ EXTERN ComputeStatus one_off(const char* biom_filename, const char* tree_filenam
                              const char* unifrac_method, bool variance_adjust, double alpha,
                              bool bypass_tips, unsigned int threads, mat_t** result);
 
+
+/* Compute UniFrac - condensed form, in-memory
+ *
+ * table <biom> a constructed BIOM object
+ * tree <BPTree> a constructed BPTree object
+ * unifrac_method <const char*> the requested unifrac method.
+ * variance_adjust <bool> whether to apply variance adjustment.
+ * alpha <double> GUniFrac alpha, only relevant if method == generalized.
+ * bypass_tips <bool> disregard tips, reduces compute by about 50%
+ * threads <uint> the number of threads to use.
+ * result <mat_t**> the resulting distance matrix in condensed form, this is initialized within the method so using **
+ *
+ * one_off_inmem returns the following error codes:
+ *
+ * okay           : no problems encountered
+ * unknown_method : the requested method is unknown.
+ * table_empty    : the table does not have any entries
+ */
+EXTERN ComputeStatus one_off_inmem(support_biom_t *table_data, support_bptree_t *tree_data,
+                                   const char* unifrac_method, bool variance_adjust, double alpha,
+                                   bool bypass_tips, unsigned int threads, mat_t** result);
+
+/* define this thing... */
```

Comment:
Do we really need to define it here?
Or would it be better as an internal function, only?

Reply:
If we move it internal (perhaps to su.cpp?), then we need to migrate the definition of ComputeStatus to avoid a circular dependency. Any concern if we move ComputeStatus to a new header?

Reply:
What I propose is we leave it here for now, on the rational that other non-exposed methods exist in api.cpp/hpp, get things working, and consider a refactor later. Does that sound reasonable? Part of the hesitation to solve immediately is it requires centralizing the currently used macros, or refactoring them into functions. Or rather, it doesn't seem like a straight forward reorganization. If you agree, I can look at it on a follow up PR. Does that sound okay?

Reply:
nm, actually i think it is important to do this as its signature contains non-C types. I'll sort it out on this PR